lib/counter_culture/helpers.rb

```diff
@@ -0,0 +1,138 @@
+module CounterCulture
+  module Helpers
+    extend ActiveSupport::Concern
+    
+    module ClassMethods
+      # this holds all configuration data
+      def after_commit_counter_cache
+        config = @after_commit_counter_cache || []
+        if superclass.respond_to?(:after_commit_counter_cache) && superclass.after_commit_counter_cache
+          config = superclass.after_commit_counter_cache + config
+        end
+        config
+      end
+
+      # called to configure counter caches
+      def counter_culture(relation, options = {})
+        unless @after_commit_counter_cache
+          include AfterCommitAction unless include?(AfterCommitAction)
+
+          # initialize callbacks only once
+          after_create :_update_counts_after_create
+          after_destroy :_update_counts_after_destroy
+          after_update :_update_counts_after_update
+
+          # we keep a list of all counter caches we must maintain
+          @after_commit_counter_cache = []
+        end
+
+        if options[:column_names] && !options[:column_names].is_a?(Hash)
+          raise ":column_names must be a Hash of conditions and column names" 
+        end
+
+        # add the current information to our list
+        @after_commit_counter_cache << Counter.new(self, {
+          :relation => relation.is_a?(Enumerable) ? relation : [relation],
+          :counter_cache_name => (options[:column_name] || "#{name.tableize}_count"),
+          :column_names => options[:column_names],
+          :delta_column => options[:delta_column],
+          :foreign_key_values => options[:foreign_key_values],
+          :touch => options[:touch]
+        })
+      end
+
+      # checks all of the declared counter caches on this class for correctnes based
+      # on original data; if the counter cache is incorrect, sets it to the correct
+      # count
+      #
+      # options:
+      #   { :exclude => list of relations to skip when fixing counts,
+      #     :only => only these relations will have their counts fixed }
+      # returns: a list of fixed record as an array of hashes of the form:
+      #   { :entity => which model the count was fixed on,
+      #     :id => the id of the model that had the incorrect count,
+      #     :what => which column contained the incorrect count,
+      #     :wrong => the previously saved, incorrect count,
+      #     :right => the newly fixed, correct count }
+      #
+      def counter_culture_fix_counts(options = {})
```

Comment:
I wanted to have more precise control rather than a bulk loop. But you're right It doesn't need to live here. Though Reconciler is the wrong place I think. 

I was originally going to make the array of counters a first class object. But there wasn't enough behaviour to warrant it as it just loops mainly. But moving this method that acts on the collection into a collection object of counters would be my preference.